pymc3/sampling.py

```diff
@@ -1,3 +1,4 @@
+from typing import Dict, List, Any, Optional
```

Comment:
Also remove unused imports to fix lint

Reply:
I'm not sure what to do about this -- these imports **are** used by mypy, AFAIK.  I don't know how the checker finds type definitions without imports.
I'm happy to do what's right here, but once again, I don't understand how 3.5 functions.

Reply:
Travis' logs say

> ++++pylint --rcfile=.pylintrc pymc3
> ************* Module pymc3.sampling
> pymc3/sampling.py:1:
[W0611(unused-import), ] Unused Optional imported from typing

Reply:
@lucianopaz At least for now I have 'fixed' this by telling pylint to ignore those imports.  I haven't been able to find an alternative solution, because I don't know otherwise how mypy is to identify the types.  I think what we have now is an acceptable expedient, unless someone can let me know of a better solution. I'll look for some advice myself.